language/solana/move-to-solana/src/stackless/module_context.rs

```diff
@@ -776,15 +809,26 @@ impl<'mm, 'up> ModuleContext<'mm, 'up> {
         }
     }
 
-    pub fn emit_rtcall_abort_raw(&self, val: u64) {
-        let thefn = self.get_runtime_function_by_name("abort");
-        let param_ty = self.llvm_cx.int_type(64);
+    pub fn emit_rtcall_abort_raw(
+        llvm_cx: &'up llvm::Context,
+        llvm_builder: &llvm::Builder,
+        llvm_module: &'up llvm::Module,
+        rtty_cx: &RttyContext,
+        val: u64,
+    ) {
```

Comment:
This and several other methods need to be used by EntrypointGenerator which has its own LLVM module that isn't included in an existing ModuleContext, therefore I tried to reuse the code by converting these methods to 'class'-level methods instead of 'instance'-level. Maybe some other refactoring is needed to have a better design and code reuse. I would appreciate suggestions how to improve this.

Reply:
Added a comment.